plugins/inputs/socketstat/socketstat.go

```diff
@@ -0,0 +1,188 @@
+package socketstat
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Socketstat is a telegraf plugin to gather indicators from established connections, using iproute2's  `ss` command.
+type Socketstat struct {
+	lister      socketLister
+	SocketProto []string
+	Timeout     internal.Duration
+}
+
+type socketLister func(proto string, Timeout internal.Duration) (*bytes.Buffer, error)
+
+const measurement = "socketstat"
+
+var defaultTimeout = internal.Duration{Duration: time.Second}
```

Comment:
Maybe move this to an `Init()` function called at construction time of the plugin.

Reply:
Can we at least move this to where the variable is used (in `init()`) so we don't work with global variables here!

Reply:
Done (added an Init() and moved this variable initialization there).